apollo-federation/src/schema/blueprint.rs

```diff
@@ -416,18 +469,191 @@ mod tests {
         );
     }
 
+    #[test]
+    fn rejects_non_root_use_of_default_query_name() {
+        let schema = Schema::parse(
+            r#"
+            schema {
+                query: MyQuery
+            }
+
+            type MyQuery {
+                f: Int
+            }
+
+            type Query {
+                g: Int
+            }
+            "#,
+            "test.graphqls",
+        )
+        .expect("parses schema");
+        let error = build_subgraph("S", &schema, true).expect_err("fails validation");
```

Comment:
I think the messages passed to [`expect()`](https://doc.rust-lang.org/std/result/enum.Result.html#method.expect) and [`expect_err()`](https://doc.rust-lang.org/std/result/enum.Result.html#method.expect_err) could be clearer. The "Recommended Message Style" from the Rust docs suggests using the word "should" and describing the reason _why_ you expect the `Result` to be `Ok` or _why_ you expect an error.

```suggestion
        .expect("The schema should parse successfully");
        let error = build_subgraph("S", &schema, true).expect_err("The invalid schema should fail validation");
```

I'd say the code reads more intuitively this way. Thoughts?

Reply:
I think that makes sense if you're going to panic and propagate an error to an end-user, but I usually keep these short in tests because I actually care about the content of the `Err` that's being unwrapped. To me, it's more of a tag/line number indicator so I know which assertion is failing when I'm developing the test.